src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionDescriptorBuilder.cs

```diff
@@ -391,13 +393,15 @@ private static AttributeRouteInfo CreateAttributeRouteInfo(AttributeRouteModel r
             {
                 actionDescriptor.AttributeRouteInfo.Template = AttributeRouteModel.ReplaceTokens(
                     actionDescriptor.AttributeRouteInfo.Template,
-                    actionDescriptor.RouteValues);
+                    actionDescriptor.RouteValues,
+                    actionDescriptor.AttributeRouteInfo.RouteTokenTransformer);
 
                 if (actionDescriptor.AttributeRouteInfo.Name != null)
                 {
                     actionDescriptor.AttributeRouteInfo.Name = AttributeRouteModel.ReplaceTokens(
                         actionDescriptor.AttributeRouteInfo.Name,
-                        actionDescriptor.RouteValues);
+                        actionDescriptor.RouteValues,
+                        actionDescriptor.AttributeRouteInfo.RouteTokenTransformer);
```

Comment:
Possible to pass this into this method rather than using `ARI` to truck it around?

Reply:
No, it has to come from `ApplicationModel`. Maybe it could live in `ControllerActionDescriptor.Properties`?

Reply:
Well `ApplicationModel ` is fine. `AttributeRouteInfo` is less fine